### ISSUE TITLE
August Dependency Updates

### ISSUE THREAD

**Issue by joewhitsitt** [uiowa/uiowa#5438]
We should try to review and merge in dependency updates on a monthly basis. Until this is automated more, here is a guide:

- Issue summary should contain a list of the dependabot pulls in scope for reference, however, the actual work should be done in a single PR to minimize composer conflicts.
- Dependencies related to our frontend build tools (not executed on the websites) are usually safe to merge as-is but might cause issue with other packages.
- As a separate PR is created with the dependency update, the related dependabot version should be closed referencing the monthly PR.
- Periodically check [locked dependencies](https://github.com/uiowa/uiowa/blob/main/README.md#locked-packages) for new versions. Make notes about why a dependency was/is locked in the repo Readme.
- Review the changelog for each version we are jumping to and the versions in-between. Make note of certain functionality in our codebase that is using the functionality for testing instructions later.
- Use https://github.com/uiowa/uiowa#updating-dependencies as a guide to test for update hooks, configuration changes and commit. This command is helpful if there are a lot of configuration changes across feature/site splits: https://github.com/uiowa/uiowa/blob/main/blt/src/Blt/Plugin/Commands/ConfigSplitCommands.php
- If a particular dependency is too big to go out along with others, split it off from the PR as a separate issue and introduce the "why" in parking lot.
- Update hooks and requires cache rebuild type updates may require off-hours deployment.
- Once this issue is completed, copy these notes to the next month's issue and introduce it in parking lot.

## Updates in-scope
- [x] https://github.com/uiowa/uiowa/pull/5533
- [x] https://github.com/uiowa/uiowa/pull/5532
- [x] https://github.com/uiowa/uiowa/pull/5531
- [x] https://github.com/uiowa/uiowa/pull/5530
- [x] https://github.com/uiowa/uiowa/pull/5517
- [x] https://github.com/uiowa/uiowa/pull/5535
- [x] https://github.com/uiowa/uiowa/pull/5515
- [x] https://github.com/uiowa/uiowa/pull/5514
- [x] https://github.com/uiowa/uiowa/pull/5513
- [x] https://github.com/uiowa/uiowa/pull/5539
- ~~https://github.com/uiowa/uiowa/pull/5511~~
- [x] https://github.com/uiowa/uiowa/pull/5510
- ~~https://github.com/uiowa/uiowa/pull/5509~~
- ~~https://github.com/uiowa/uiowa/pull/5508~~
- [x] https://github.com/uiowa/uiowa/pull/5419
- [x] https://github.com/uiowa/uiowa/pull/5323
- [x] https://github.com/uiowa/uiowa/pull/5215

## Possible locked dependency updates

- cshs
- menu_link_weight
- reroute_email

## Separate issues

- https://github.com/uiowa/uiowa/pull/5324 will be addressed in #5467
- https://github.com/uiowa/uiowa/pull/5205 will be addressed in #5332

**Comment by pyrello** [comment]
Making a note that this PR is going to require extra work and testing: #5324. This is because the new version of the dependency changes a setting that introduces a lot more deprecation errors. Each of those errors will need to be fixed and we'll need to test that the corresponding functionality still works.

**Comment by briand44** [comment]
Setting up initial list and PR is a good pairing opportunity.

**Comment by joewhitsitt** [comment]
Going to start looking at this later today. Will start huddle.

**Comment by joewhitsitt** [comment]
Waiting for a security update to allow merging babel bumps.

Going to continue adding to PR the rest of the updates

**Comment by joewhitsitt** [comment]
Related to the menu_per_role update, created an issue to see if we can leverage settings instead of custom code: https://github.com/uiowa/uiowa/issues/5537

**Comment by joewhitsitt** [comment]
Not sure #5511 can be upgraded without looking at upgrading https://www.drupal.org/project/fontawesome

**Comment by joewhitsitt** [comment]
I've updated but haven't looked too much into cshs, menu_link_weight and reroute_email yet.

**Comment by bspeare** [comment]
When this is ready for review, I wouldn't mind pairing with someone on testing this to learn the process.

**Comment by joewhitsitt** [comment]
I've updated the test instructions. Fairly straight forward I hope. If you fully test, you will get bad config exports, but you would get the same bad config exports on main. Use your best judgement. Alternatively try this branch on a few sites in DEV to see if the config sticks.

None of these modules are changing functionality in big ways so don't sweat too much.

**Comment by joewhitsitt** [comment]
If y'all want to wait on this, I am fine with that.

**Comment by briand44** [comment]
Does it complicate things more if we wait or no?

**Comment by joewhitsitt** [comment]
- no security updates within this.
- no fixes to known issues on PROD.
- layout_builder_restrictions touches a lot of config so some things might be updated as more is merged into main.
- composer lock will get messed up the longer we wait.
- most if not all are prerequisites to 9.4

**Comment by joewhitsitt** [comment]
Created #5584
